neuralop/layers/transformer_block.py

```diff
@@ -0,0 +1,282 @@
+            **kwargs,
+    ):
+        super().__init__()
+        self.in_channels = in_channels
+        self.out_channels = out_channels
+        self.hidden_channels = hidden_channels
+        self.num_heads = num_heads
+        self.head_n_channels = head_n_channels
+        self.n_layers = n_layers
+        self.use_mlp = use_mlp
+        self.mlp_dropout = mlp_dropout
+        self.mlp_expansion = mlp_expansion
+        self.non_linearity = non_linearity
+        self.norm = norm
+
+        self.lifting = nn.Linear(self.in_channels, self.hidden_channels) \
+            if self.in_channels != self.hidden_channels else nn.Identity()
+
+        self.to_out = nn.Linear(self.hidden_channels, self.out_channels) \
+            if self.hidden_channels != self.out_channels else nn.Identity()
+
+        self.attention_norms = nn.ModuleList([get_normalization(self.norm, self.hidden_channels) for _ in range(self.n_layers)])
+        self.attention_layers = nn.ModuleList([
+                                    AttentionKernelIntegral(
+                                        in_channels=self.hidden_channels,
+                                        out_channels=self.hidden_channels,
+                                        n_heads=self.num_heads,
+                                        head_n_channels=self.head_n_channels,
+                                        project_query=True)
+                                    for _ in range(self.n_layers)])
+
+        if self.use_mlp:
+            self.mlp_norms = nn.ModuleList([get_normalization(self.norm, self.hidden_channels) for _ in range(self.n_layers)])
+            self.mlp_layers = nn.ModuleList([
+                                    MLPLinear([self.hidden_channels,
+                                       int(self.hidden_channels * self.mlp_expansion),
+                                       self.hidden_channels],
+                                       dropout=self.mlp_dropout)
+                                for _ in range(self.n_layers)])
+
+    def forward(self,
+                u,
+                pos,
+                pos_emb_module=None,
+                **kwargs):
+        """
+        Encode the input function u using the Transformer Encoder Block.
+
+        Parameters:
+            u: torch.Tensor, input tensor of shape [batch_size, num_grid_points, channels]
+            pos: torch.Tensor, grid point coordinates of shape [batch_size, num_grid_points, channels]
+            pos_emb_module: nn.Module, positional embedding module, by default None
+
+        """
+        u = self.lifting(u)
+        for l in range(self.n_layers):
+            u_attention_skip = u
+            u = self.attention_layers[l](u_src=normalize(u, self.attention_norms[l]),
+                                     pos_src=pos,
+                                     positional_embedding_module=pos_emb_module,
+                                     **kwargs)
+            u = u + u_attention_skip
```

Comment:
Have you noticed any empirical improvements from having this skip connection be a pointwise linear (not MLP) as opposed to identity connection? If so, it could be interesting to supplement with this functionality.